File: 643.py
class Solution:
    def findMaxAverage(self, nums: list[int], k: int) -> float:
        n = len(nums)
        curr = sum(nums[:k])
        currMax = curr
        for i in range(1, n - k + 1):
            curr += nums[i + k - 1] - nums[i - 1]
            currMax = max(currMax, curr)
        return currMax / k
        
if __name__ == '__main__':
    nums = list(map(int, input().split()))
    k = int(input())
    s = Solution()
    print(s.findMaxAverage(nums, k))
    

File: 643.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        int curr = 0;
        for(int i = 0; i < k; i++)
            curr += nums[i];
        int currMax = curr;
        for(int i = 1; i <= n - k; i++){
            curr += nums[i + k - 1] - nums[i - 1];
            currMax = max(currMax, curr);
        }
        return (double)currMax / k;
    }
};

int main(){
    int n;
    cin >> n;
    vector<int> nums(n);
    for(int i = 0; i < n; i++)
        cin >> nums[i];
    int k;
    cin >> k;
    Solution s;
    cout << s.findMaxAverage(nums, k) << endl;
    return 0;
}